diff --git a/selfhost/llvm/text/function_calls.slg b/selfhost/llvm/text/function_calls.slg
index 0e2a46a2..aed5f851 100644
--- a/selfhost/llvm/text/function_calls.slg
+++ b/selfhost/llvm/text/function_calls.slg
@@ -709,7 +709,10 @@ emitFunctionCalls expressionIndex: Int, functionIndex: Int, expressionStart: Int
                                             "  %v$(expressionIndex)_runtime_integer = " -> print
                                             runtimeArgument -> integerNodeSigned(context, state)
                                                 -> if { "sext" -> print } else { "zext" -> print }
-                                            " i$(runtimeArgumentWidth!) %v$(runtimeArgumentIndex) to i32" -> println
+                                            " i$(runtimeArgumentWidth!) " -> print
+                                            runtimeArgument.kind == 5 and (context -> writeParameterReference(functionIndex, runtimeArgument.symbol))
+                                                -> unless { "%v$(runtimeArgumentIndex)" -> print }
+                                            " to i32" -> println
                                         }
                                     runtimeArgumentWidth! == 64 -> if {
                                         expression.symbol == -116 -> if { "  call void @sollang_runtime_eprint_i64(i64 " -> print } else { "  call void @sollang_runtime_print_i64(i64 " -> print }
